packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -208,7 +208,12 @@ export const run = async (context: $TSContext, resourceDefinition: $TSObject, re
     await buildOverridesEnabledResources(context, resources);
 
     // print cdk migration warning
-    await printCdkMigrationWarning(context);
+    try{
+      await printCdkMigrationWarning(context);
+    }catch(error){
+      printer.warn(`Amplify detector failed with error ${error}`);
+      printer.warn(`To migrate to AWS CDK V2 : https://docs.amplify.aws/cli/migration/aws-cdk-migration/`);
+    }
```

Comment:
if we want to suppress errors from detector (which isn't bad idea) we should:
- move try-catch inside `printCdkMigrationWarning`
- don't print anything - if it's false positive we should just be quiet.